tests/test_cli.py

```diff
@@ -0,0 +1,66 @@
+import pytest
+import sys
+
+import os
+from komodo.cli import cli_main
+from tests import _get_test_root
+import shutil
+
+
+@pytest.mark.parametrize(
+    "args",
+    [
+        (
+            "--download",
+            "--build",
+            "--install",
+            "--prefix",
+            "prefix",
+            "--cache",
+            "cache",
+            "--tmp",
+            "tmp",
+            "--release",
+            "nominal_release",
+            "--pip",
+            "/bin/true",
+            "--extra-data-dirs",
+            os.path.join(_get_test_root(), "data/cli"),
+            os.path.join(_get_test_root(), "data/cli/hackres"),
+            "--locations-config",
+            os.path.join(_get_test_root(), "data/cli/locations.yml"),
+            os.path.join(_get_test_root(), "data/cli/nominal_release.yml"),
+            os.path.join(_get_test_root(), "data/cli/nominal_repository.yml"),
+        ),
+    ],
+)
+def test_main(args, tmpdir):
+    shutil.copytree(
+        os.path.join(_get_test_root(), "data/cli/hackres"),
+        os.path.join(tmpdir, "hackres"),
+    )
+
+    sys.argv = [
+        "kmd",
+        "--workspace",
+        str(tmpdir),
+    ]
+
+    # Komodo expects the renamer from util-linux. On Debian/Ubuntu, this
+    # renamer has been renamed to rename.ul.
+    if os.path.exists("/usr/bin/rename.ul"):
+        sys.argv.extend(["--renamer", "/usr/bin/rename.ul"])
+
+    sys.argv.extend([arg for arg in args])
```

Comment:
`sys.argv.extend(args)`?

In general I think modifying `sys.argv` at runtime is not a sign of quality software. Then again, this is komodo, so nevermind.

Reply:
You're right, it's bad. I'll change this so that I use `_main` instead, and then do argparse testing elsewhere referencing the parser explicitly.